lighthouse-core/computed/critical-request-chains.js

```diff
@@ -66,6 +66,68 @@ class CriticalRequestChains {
     return ['VeryHigh', 'High', 'Medium'].includes(request.priority);
   }
 
+  /**
+   * Create a tree of critical requests.
+   * @param {LH.Artifacts.NetworkRequest} mainResource
+   * @param {LH.Gatherer.Simulation.GraphNode} graph
+   * @return {LH.Artifacts.CriticalRequestNode}
+   */
+  static extractChainsFromGraph(mainResource, graph) {
+    /** @type {LH.Artifacts.CriticalRequestNode} */
+    const rootNode = {};
+
+    /**
+     * @param {LH.Artifacts.NetworkRequest[]} path
+     */
+    function addChain(path) {
+      let currentNode = rootNode;
+
+      for (const record of path) {
+        if (!currentNode[record.requestId]) {
+          currentNode[record.requestId] = {
+            request: record,
+            children: {},
+          };
+        }
+
+        currentNode = currentNode[record.requestId].children;
+      }
+    }
+
+    // By default `traverse` will discover nodes in BFS-order regardless of dependencies, but
+    // here we need traversal in a topological sort order. We'll visit a node only when its
+    // dependencies have been met.
+    const seenNodes = new Set();
+    /**
+     * @param {LH.Gatherer.Simulation.GraphNode} node
+     */
+    function getNextNodes(node) {
+      return node.getDependents().filter(n => {
+        const hasSeenAllDependants = n.getDependencies().every(d => seenNodes.has(d));
+        return hasSeenAllDependants;
```

Comment:
```suggestion
        const hasSeenAllDependencies = n.getDependencies().every(d => seenNodes.has(d));
        return hasSeenAllDependencies;
```
Or maybe just remove this temp var?